packages/pyright-internal/src/tests/samples/semantic_highlighting/parameters.py

```diff
@@ -0,0 +1,11 @@
+class C:
+    def __init__(self, x):
+        self.x = x
+
+def f(x, y: int):
+    def g(x):
+        return x * y
+    z = 2 + x
```

Comment:
fyi the fact that `z` gets no token is a bug (#180)

Reply:
not saying you have to fix it, just letting you know

Reply:
No problem, I think I can fix it, maybe in a separate pull request or would it be okay to do it in this one?

Reply:
Also, should I give `self` and `cls` some sort of special token/modifier? I think it would be good since you usually want to distinguish them from regular parameters. Does pylance do something similar or just threats it as a regular parameter?

Reply:
> No problem, I think I can fix it, maybe in a separate pull request or would it be okay to do it in this one?

maybe a separate MR, makes it cleaner for the changelog i think

> Also, should I give `self` and `cls` some sort of special token/modifier? I think it would be good since you usually want to distinguish them from regular parameters. Does pylance do something similar or just threats it as a regular parameter?

pylance just treats them as variables or parameters (my theme gives both of them the same colour so i'm not sure which one). basedpyright does deviate with `cls` tho where it's treated as a type instead (which was originally unintended but i figured it was better so i decided to just stick with it). i assume this PR would change that functionality to match what pylance does, which i dont mind. maybe add a test for it tho

Reply:
I was asking more to know if there was some sort expectations about it. I actually tried to implement it but i find it to cumbersome to get right. I do like how `cls` is colored as a type, even if it was not intentional. I think I just leave it as is, to treat `self` as a regular parameter and let `cls` do its thing.

Reply:
looks like it now has a different colour in the signature to the body:

![image](https://github.com/DetachHead/basedpyright/assets/57028336/45272f40-51a9-4cd8-b554-f2b66a889864)

maybe it's best to just make it always a parameter

Reply:
Yeah, it might be better for it to just be a parameter since it seems that the priority of the semantic tokens is highly dependent on the editor/color scheme.